packages/x-data-grid/src/constants/cssVariables.ts

```diff
@@ -0,0 +1,173 @@
+// NOTE: Breakpoints can't come from the theme because we need access to them at
+// initialization time and media-queries can't use CSS variables. For users with
+// custom breakpoints, we might want to provide a way to configure them globally
+// instead of through the theme.
+const breakpoints = {
+  values: {
+    xs: 0, // phone
+    sm: 600, // tablet
+    md: 900, // small laptop
+    lg: 1200, // desktop
+    xl: 1536, // large screen
+  },
+  up: (key: any) => {
+    const values = breakpoints.values as any;
+    const value = typeof values[key] === 'number' ? values[key] : key;
+    return `@media (min-width:${value}px)`;
+  },
+};
+
+const keys = {
+  spacingUnit: '--DataGrid-t-spacing-unit',
+
+  /* Variables */
+  colors: {
+    border: {
+      base: '--DataGrid-t-colors-border-base',
+    },
+    foreground: {
+      base: '--DataGrid-t-colors-foreground-base',
+      muted: '--DataGrid-t-colors-foreground-muted',
+      accent: '--DataGrid-t-colors-foreground-accent',
+      disabled: '--DataGrid-t-colors-foreground-disabled',
+    },
+    background: {
+      base: '--DataGrid-t-colors-background-base',
+      overlay: '--DataGrid-t-colors-background-overlay',
+      backdrop: '--DataGrid-t-colors-background-backdrop',
+    },
+    interactive: {
+      hover: '--DataGrid-t-colors-interactive-hover',
+      hoverOpacity: '--DataGrid-t-colors-interactive-hover-opacity',
+      focus: '--DataGrid-t-colors-interactive-focus',
+      focusOpacity: '--DataGrid-t-colors-interactive-focus-opacity',
+      disabled: '--DataGrid-t-colors-interactive-disabled',
+      disabledOpacity: '--DataGrid-t-colors-interactive-disabled-opacity',
+      selected: '--DataGrid-t-colors-interactive-selected',
+      selectedOpacity: '--DataGrid-t-colors-interactive-selected-opacity',
+    },
+  },
+  header: {
+    background: {
+      base: '--DataGrid-t-header-background-base',
+    },
+  },
```

Comment:
@KenanYusuf I have calqued this one on `cell.background.pinned` but please confirm if you want it this way.

Reply:
Yep looks good 👍